cardano-cli/src/Cardano/CLI/Shelley/Run/Transaction.hs

```diff
@@ -447,6 +302,215 @@ runTransactionCmd cmd =
 -- Building transactions
 --
 
+runTxBuildCmd
+  :: IsCardanoEra era
+  => CardanoEra era
```

Comment:
LGTM, however lets use `AnyCardanoEra` here and add some haddock annotations to the parameters where it's not obvious what they are e.g `Maybe Word`.  The haddocks don't have to be overly descriptive, short succinct ones will do.